### PR TITLE
Initial support for AniSkip

### PR DESCRIPTION
**Pull request by Diegopyl1209**
POV psychopath:

https://user-images.githubusercontent.com/80992641/190887539-0ea0ce56-fe17-483e-a643-90b51a560990.mp4

POV normal person:

https://user-images.githubusercontent.com/80992641/190887570-8e94b72e-8f8e-4332-8503-f2b59c0b96e9.mp4

### CODE REVIEW

**Review by jmir1 — APPROVED** [review]

**Comment by Diegopyl1209** [comment]
AniSkip Stuff:

Settings:
Screenshot_20220919-182255_Aniyomi

Without Netflix Style:

https://user-images.githubusercontent.com/80992641/191123820-cc8f5103-7abf-4089-ba4b-78bb690d4137.mp4

With Netflix Style:

https://user-images.githubusercontent.com/80992641/191124300-6942b211-22fd-4fd9-a10c-75d05aa22e72.mp4

Whit AutoSkip Enabled:


https://user-images.githubusercontent.com/80992641/191123888-b7716fe0-ab8b-4410-be43-6266acc2d9bf.mp4

**Comment by Diegopyl1209** [comment]
emotiguy-bola-amarilla

**Review by jmir1 — COMMENTED** [review]
I think the "manual" skip button should be replaced by the aniskip button. It looks very cluttered with both of them visible.
Otherwise, it's looking pretty good!
Code could be a bit cleaner but we'll get to that.

**Comment by Diegopyl1209** [comment]
https://user-images.githubusercontent.com/80992641/191586101-f50d981f-207f-427e-a883-aff709bc44fc.mp4

File: app/src/main/java/eu/kanade/tachiyomi/util/AniSkipApi.kt [hunk 1]
```diff
@@ -0,0 +1,149 @@
...
+        val payload = buildJsonObject {
+            put("query", query)
+        }
+        val response = client.newCall(
+            POST(
+                "https://graphql.anilist.co",
+                body = payload.toString().toRequestBody(jsonMime),
+            ),
+        ).execute()
+        // i need to change this
```
Comment (jmir1) [inline app/src/main/java/eu/kanade/tachiyomi/util/AniSkipApi.kt:57]: then maybe do that ^^

File: app/src/main/java/eu/kanade/tachiyomi/ui/player/PlayerPresenter.kt [hunk 2]
```diff
@@ -585,6 +589,36 @@ class PlayerPresenter(
...
+            malId = when (service) {
+                is MyAnimeList -> track.remoteId
+                is Anilist -> AniSkipApi().getMalIdFromAL(track.remoteId)
+                else -> null
+            }
+            val duration = view?.player?.duration ?: return null
+            return when (malId) {
+                null -> null
+                else -> AniSkipApi().getResult(malId!!.toInt(), episodeNumber, duration.toLong())
+            }
```
Comment (jmir1) [inline app/src/main/java/eu/kanade/tachiyomi/ui/player/PlayerPresenter.kt:618]: ???

```suggestion
            return malId?.let {
                AniSkipApi().getResult(it.toInt(), episodeNumber, duration.toLong())
            }
```

File: app/src/main/java/eu/kanade/tachiyomi/util/AniSkipApi.kt [hunk 3]
```diff
@@ -0,0 +1,149 @@
+package eu.kanade.tachiyomi.util
+
+import android.annotation.SuppressLint
+import android.util.Log
```
Comment (jmir1) [inline app/src/main/java/eu/kanade/tachiyomi/util/AniSkipApi.kt:4]: just remove all of the `Log`s pls

File: app/src/main/java/eu/kanade/tachiyomi/util/AniSkipApi.kt [hunk 4]
```diff
@@ -0,0 +1,149 @@
...
+            val skipTime = when (skipType) {
+                SkipType.ed -> aniSkipResponse.first { it.skipType == SkipType.ed }.interval
+                SkipType.op -> aniSkipResponse.first { it.skipType == SkipType.op }.interval
+                SkipType.recap -> aniSkipResponse.first { it.skipType == SkipType.recap }.interval
+                SkipType.mixedOp -> aniSkipResponse.first { it.skipType == SkipType.mixedOp }.interval
+            }
+            if (waitingTime > -1) {
+                if (waitingTime > 0) {
+                    launchUI {
+                        playerControls.binding.controlsSkipIntroBtn.text = activity.getString(R.string.player_aniskip_dontskip) + " " + waitingTime
```
Comment (jmir1) [inline app/src/main/java/eu/kanade/tachiyomi/util/AniSkipApi.kt:93]: i think it looks a bit weird when there's just a normal number next to the text.
the user can't really understand that it's a countdown.
it probably needs a little more space and a different text style

**Review by jmir1 — CHANGES_REQUESTED** [review]
https://youtu.be/Mg5HOnq7zD0?t=4

File: app/src/main/java/eu/kanade/tachiyomi/util/AniSkipApi.kt (hunk 4 above)
Comment (Diegopyl1209) [reply app/src/main/java/eu/kanade/tachiyomi/util/AniSkipApi.kt]: https://user-images.githubusercontent.com/80992641/191633075-3f0f3351-3148-4f10-9a68-02025cfcc7db.mp4

Comment (jmir1) [reply app/src/main/java/eu/kanade/tachiyomi/util/AniSkipApi.kt]: I think that's better

**Comment by jmir1** [comment]
Ok, last two things: 
1. If autoskip is enabled, the +85s button should be completely removed because it honestly serves no purpose.
2. When autoskipping, it should play the skip animation that also plays when you press +85s right now. The text should be "Skipped Opening"/"Skipped Ending" etc. Then you could also remove that toast.

**Comment by Diegopyl1209** [comment]
> 


https://user-images.githubusercontent.com/80992641/191838227-a3f97c8f-7163-4b63-86a6-a8ed7445f517.mp4

I made the +85 button show even if you have aniskip activated only when it doesn't find the opening, otherwise it won't appear

**Comment by jmir1** [comment]
Yes that is what I had in mind.
LGTM! Thanks for your hard work :pray:

**Review by jmir1 — APPROVED** [review]

**Comment by GoluLiliput** [comment]
When will the AniSkip be implemented in public version?

**Comment by Spacellary** [comment]
I'm crying here this is beautiful my friends, can't wait for this!

**Pull request merged**